crates/ncmdump/src/ncmdump.rs

```diff
@@ -55,6 +84,39 @@ where
     key_box: Vec<usize>,
 }
 
+impl NcmInfo {
+    pub fn from(raw_info: RawNcmInfo) -> Self {
+        Self {
+            name: raw_info.name,
+            id: raw_info.id.get_id().unwrap(),
+            album: raw_info.album,
+            artist: raw_info.artist.into_iter().map(|(name, id)| (name, id.get_id().unwrap())).collect(),
+            bitrate: raw_info.bitrate.get_id().unwrap(),
+            duration: raw_info.duration.get_id().unwrap(),
+            format: raw_info.format,
+            mv_id: match raw_info.mv_id {
+                Some(id) => id.get_id(),
+                None => None,
+            },
+            alias: raw_info.alias,
+        }
+    }
+}
+
+impl NcmId {
+    pub fn get_id(self) -> Option<u64> {
+        match self {
+            NcmId::String(s) => {
+                if s.is_empty() {
+                    return None;
+                }
+                s.parse().ok()
+            },
+            NcmId::Integer(num) => Some(num),
+        }
+    }
+}
```

Comment:
perhaps we should resolve error by better way. For example use `Resule` and `Errors`.

Reply:
```suggestion
impl NcmId {
    pub fn get_id(self) -> Option<u64> {
        match self {
            NcmId::String(s) => {
                if s.is_empty() {
                    return None;
                }
                s.parse().ok()
            },
            NcmId::Integer(num) => Some(num),
        }
    }
}
```

Reply:
I don't fully understand your mind, can u explain it to me?

Reply:
The `ncmId::String("")` will make a panic.

Reply:
u mean this function should panic when parsing NcmId::String("")?
i am not a native english speaker, apologies for my poor english😖